homeassistant/components/stream/__init__.py

```diff
@@ -50,11 +50,7 @@ def request_stream(hass, stream_source, *, fmt="hls", keepalive=False, options=N
         options = {}
 
     # For RTSP streams, prefer TCP
-    if (
-        isinstance(stream_source, str)
-        and stream_source[:7] == "rtsp://"
-        and not options
-    ):
+    if isinstance(stream_source, str) and stream_source[:7] == "rtsp://":
```

Comment:
why drop "and not options" ?

Reply:
because if we specify the `rtsp_transport` then `options` will not be empty. And i still want to add the `rtsp_flags` and `stimeout`.

Reply:
For the time being I didn't want to change the handling of these 2 parameters. I am not sure if we should keep them fixed or allow the user to specify them (while having a default value for them).